### PR TITLE
✨Comp-backend override resources with selected hardware

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?
In billable products, this PR ensures the service resources are set according to the selected EC2 instance type with a safety margin of 1GiB for now as it seems that the dask-sidecar *sees* a bit less than what is expected.

### details
- [x] ec2_instances client side implementation in service-library
- [x] director-v2 gets ec2 resources based on pricing unit if available, updates projects_nodes accordingly (NOTE: this should move to the webserver)
- [x] some cleanup and codeclimate linting
### bonus
- [x] webserver correctly fills *projects_nodes* table when a project is created from a service

### CODE REVIEW

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/clusters.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+from ....rabbitmq import RabbitMQRPCClient
+
+
+async def get_or_create_cluster(
+    client: RabbitMQRPCClient, *, user_id: UserID, wallet_id: WalletID | None
+) -> OnDemandCluster:
+    on_demand_cluster: OnDemandCluster = await client.request(
+        CLUSTERS_KEEPER_RPC_NAMESPACE,
+        RPCMethodName("get_or_create_cluster"),
+        timeout_s=300,
```
Comment (pcrespov) [inline packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/clusters.py:16]: Q: I was wondering the other day about this number here. How do you determine it? Any hints?

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/ec2_instances.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+from models_library.api_schemas_clusters_keeper.ec2_instances import EC2InstanceType
+from models_library.rabbitmq_basic_types import RPCMethodName
+
+from ....rabbitmq import RabbitMQRPCClient
+
+
+async def get_instance_type_details(
+    client: RabbitMQRPCClient, *, instance_type_names: set[str]
+) -> list[EC2InstanceType]:
+    instance_types: list[EC2InstanceType] = await client.request(
```
Comment (pcrespov) [inline packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/ec2_instances.py:11]: Q: The issue of the exceptions is still fully unsolved. AFAIK this function can only raise one type of exception, right? Perhpaps we can document it and even expose as `rpc_interfaces.errors`?

**Review by pcrespov — APPROVED** [review]

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/ec2_instances.py (hunk 2 above)
Comment (sanderegg) [reply packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/ec2_instances.py:11]: I know of someone that always says: *one step at a time*
good point, will add it

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/clusters.py (hunk 1 above)
Comment (sanderegg) [reply packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/clusters.py:16]: through unit testing. but I will put a constant

**Review by matusdrobuliak66 — APPROVED** [review]
:+1:

**Review by GitHK — APPROVED** [review]

File: packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/clusters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sanderegg) [reply packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/clusters_keeper/clusters.py]: yes the push just happened now...

File: services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/dask_scheduler.py [hunk 3]
```diff
@@ -182,21 +182,29 @@ async def _stop_tasks(
...
                 ]
-            )
+                await asyncio.gather(
+                    *[
+                        client.release_task_result(t.job_id)
+                        for t in tasks_with_no_worker
+                        if t.job_id
+                    ]
+                )
+        except ComputationalBackendOnDemandNotReadyError:
```
[the remark this answers is not in the document]
Comment (sanderegg) [reply services/director-v2/src/simcore_service_director_v2/modules/comp_scheduler/dask_scheduler.py:205]: yes

**Pull request merged**